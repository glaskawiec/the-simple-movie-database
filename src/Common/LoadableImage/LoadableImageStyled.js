import styled from 'styled-components';

const LoadableImageStyled = styled.img`
  background-color: #dbdbdb;
  display: block;
  max-width: 100%;
  height: 100%;
  margin: auto;
  cursor: ${props => (props.onClick ? 'pointer' : 'default')};
  
  @media (max-width: 768px) {
    flex: 100%;
    max-width: 100%;
    height: auto;
  }
`;

export default LoadableImageStyled;
