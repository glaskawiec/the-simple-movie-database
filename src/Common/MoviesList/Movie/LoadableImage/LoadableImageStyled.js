import styled from 'styled-components';

const LoadableImageStyled = styled.img`
  background-color: #dbdbdb;
  display: block;
  max-width: 100%;
  height: 100%;
  margin: auto;
  
  @media (max-width: 768px) {
    flex: 100%;
    max-width: 100%;
    height: auto;
  }
`;

export default LoadableImageStyled;
