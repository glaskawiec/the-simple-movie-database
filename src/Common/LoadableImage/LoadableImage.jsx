import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import noImageDesktop from '../../resources/images/no-image-desktop.svg';
import noImageMobile from '../../resources/images/no-image-mobile.svg';
import LoadableImageStyled from './LoadableImageStyled';

const LoadableImage = React.memo(({ src, onClick }) => {
  const isMobile = window.innerWidth <= 768;
  const noImageSrc = isMobile ? noImageMobile : noImageDesktop;
  const [showedImageSrc, setShowedImageSrc] = useState(noImageSrc);

  useEffect(() => {
    if (!src) {
      return () => ({});
    }
    const img = new Image();
    img.onload = () => setShowedImageSrc(src);
    img.onerror = () => setShowedImageSrc(noImageSrc);
    img.src = src;
    return () => {
      img.onload = null;
      img.onerror = null;
    };
  });
  return <LoadableImageStyled onClick={onClick} src={showedImageSrc} />;
});

LoadableImage.propTypes = {
  src: PropTypes.string,
  onClick: PropTypes.func,
};

LoadableImage.defaultProps = {
  src: null,
  onClick: null,
};

export default LoadableImage;
