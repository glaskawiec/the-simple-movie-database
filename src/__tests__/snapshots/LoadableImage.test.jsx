import React from 'react';
import renderer from 'react-test-renderer';
import LoadableImage from '../../common/loadableImage/LoadableImage';

test('LoadableImage snapshot', () => {
  const tree = renderer
    .create(<LoadableImage />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
