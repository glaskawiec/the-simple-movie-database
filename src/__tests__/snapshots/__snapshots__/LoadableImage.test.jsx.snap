// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadableImage snapshot 1`] = `
<img
  className="sc-bdVaJa jCoEWt"
  onClick={null}
  src="no-image-desktop.svg"
/>
`;
